docs/src/architecture.md

```diff
@@ -1,30 +1,166 @@
+    where this topic was covered in an exhaustive manner.
+
+
+## Kubernetes architecture
+
+Kubernetes natively provides the possibility to span across separate physical
+locations - also known as data centers, failure zones, or more frequently
+**availability zones** - connected to each other via redundant, low-latency,
+private network connectivity.
+
+Being a distributed system, the recommended minimum number of availability
+zones for a Kubernetes cluster is three (3), providing a way for the control
+plane to be resilient over the failure of a single zone.
+For details, please refer to
+["Running in multiple zones"](https://kubernetes.io/docs/setup/best-practices/multiple-zones/).
+This means that **each data center is active at any time** and can run workloads
+simultaneously.
+
+!!! Note
+    Most of the public Cloud Providers' managed Kubernetes services already
+    provide 3 or more availability zones in each region.
+
+### Multi-availability zone Kubernetes clusters
+
+The multi-availability zone Kubernetes architecture, having three (3) or more
+zones, is the one that we recommend for PostgreSQL usage.
+This scenario is typical of Kubernetes services managed by Cloud Providers.
+
+![Kubernetes cluster spanning over 3 independent data centers](./images/k8s-architecture-3-az.png)
+
+Such an architecture enables the CloudNativePG operator to control the full
+lifecycle of a `Cluster` resource within a single Kubernetes cluster, by
+treating all the availability zones as active: this includes, among the others,
+[scheduling](scheduling.md) the workloads in a declarative manner (based on
+affinity rules, tolerations and node selectors), automated failover,
+self-healing, and updates.
+
+Please refer to the ["PostgreSQL architecture"](#postgresql-architecture)
+section below for details on how you can design your PostgreSQL clusters within
+the same Kubernetes cluster through shared-nothing deployments at the storage,
+worker node, and availability zone levels.
+
+Moreover, you can take advantage of additional [Kubernetes clusters](#deployments-across-kubernetes-clusters),
+by using them to host "passive" PostgreSQL replica clusters. This is to be
+intended primarily for DR, read-only operations, or cross-region availability -
+even though failovers and promotions in this case must be done manually.
+
+![Example of a multiple Kubernetes cluster architecture distributed over 3 regions each with 3 independent data centers](./images/k8s-architecture-multi.png)
+
+### Single availability zone Kubernetes clusters
+
+In case your Kubernetes cluster has only one availability zone, CloudNativePG
+still provides you with a lot of features to improve HA and DR results for your
+PostgreSQL databases, by pushing out the single point of failure (SPoF) to the
+level of the zone as much as possible.
```

Comment:
not sure if this is a bit technical for what we want to express

Reply:
I think this is actually crucial in terms of risk management - which this page ultimately is about (without mentioning it). The SPoF is definitely the data center. However, if users end up placing all Postgres instances on the same storage, for example, that becomes the SPoF for that Postgres database.